### PR TITLE
#1 - Hash file saving

### PR DESCRIPTION
**Pull request by olegmoz**

### CODE REVIEW

**Comment by 0crat** [comment]
This pull request [#11](https://github.com/artipie/nuget-adapter/issues/11) is assigned to @iakunin[/z](https://www.0crat.com/u/iakunin), here is [why](https://www.0crat.com/footprint/CT2E6TK9B/e03a982b-5643-4817-b0c8-10149064b178); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @g4s8[/z](https://www.0crat.com/u/g4s8) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

File: src/test/java/com/artpie/nuget/HashTest.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        );
+    }
+
+    @Test
+    void shouldSave() {
+        final String id = "abc";
+        final String version = "0.0.1";
+        final Hash hash = new Hash(HashCode.fromString("0123456789abcdef"));
+        hash.save(this.storage, new PackageIdentity(id, version));
+        final Key.From key = new Key.From(id, version, "abc.0.0.1.nupkg.sha512");
```
Comment (iakunin) [inline src/test/java/com/artpie/nuget/HashTest.java]: @olegmoz `key` variable can be inlined

File: src/test/java/com/artpie/nuget/PackageIdentityTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Tests for {@link PackageIdentity}.
+ *
+ * @since 0.1
+ */
+public class PackageIdentityTest {
+
+    @Test
+    void shouldGenerateHashKey() {
+        final PackageIdentity identity = new PackageIdentity("Newtonsoft.Json", "12.0.3");
```
Comment (iakunin) [inline src/test/java/com/artpie/nuget/PackageIdentityTest.java]: @olegmoz `identity` variable can be inlined

File: src/main/java/com/artpie/nuget/PackageIdentity.java [hunk 3]
```diff
@@ -55,13 +55,31 @@ public PackageIdentity(final String id, final String version) {
         this.version = version;
     }
 
+    /**
+     * Get key for hash file.
+     *
+     * @return Key to hash file.
+     */
+    public Key hashKey() {
+        final String name = String.format("%s.%s.nupkg.sha512", this.idLowerCase(), this.version);
```
Comment (iakunin) [inline src/main/java/com/artpie/nuget/PackageIdentity.java]: @olegmoz `name` variable can be inlined

File: src/main/java/com/artpie/nuget/Hash.java [hunk 4]
```diff
@@ -25,20 +25,38 @@
...
     /**
-     * Saves hash to storage.
+     * Saves hash to storage as base64 string.
      *
      * @param storage Storage to use for saving.
      * @param identity Package identity.
      */
     public void save(final BlockingStorage storage, final PackageIdentity identity) {
-        throw new UnsupportedOperationException("Not implemented");
+        final byte[] bytes = Base64.getEncoder().encode(this.value.asBytes());
```
Comment (iakunin) [inline src/main/java/com/artpie/nuget/Hash.java]: @olegmoz `bytes` variable can be inlined

**Comment by iakunin** [comment]
@olegmoz thanks

**Comment by olegmoz** [comment]
@iakunin Fixed all issues you mentioned, thanks

**Comment by yegor256** [comment]
@olegmoz thanks

**Comment by 0crat** [comment]
Job audit: `ARC` didn't merge or close PR

**Comment by 0crat** [comment]
Job was finished in 19 hours, bonus for fast delivery is possible (see [§36](http://www.zerocracy.com/policy.html#36))

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @g4s8[/z](https://www.0crat.com/u/g4s8)

**Comment by 0crat** [comment]
Order was finished: +20 point(s) just awarded to @iakunin[/z](https://www.0crat.com/u/iakunin)

**Pull request merged**